api/manifest.go

```diff
@@ -40,58 +38,96 @@ func apiManifestHandler(w http.ResponseWriter, r *http.Request) {
 	w.Write(manifest)
 }
 
-func getManifest(ctx context.Context, manifestAPI manifest.API, sha string, paths []string) (string, []byte, error) {
-	if sha == "" {
-		sha = "latest"
-	}
-	fetchedSHA := sha
+func getManifest(log shared.Logger, manifestAPI manifest.API, sha string, paths []string) (string, []byte, error) {
+	mc := manifestAPI.NewMemcache(time.Hour * 48)
+	// Shorter expiry for latest SHA, to keep it current.
+	latestMC := manifestAPI.NewMemcache(time.Minute * 5)
+
+	var fetchedSHA string
 	var body []byte
-	cached, err := memcache.Get(ctx, manifestCacheKey(sha))
 
-	if err != nil && err != memcache.ErrCacheMiss {
-		return "", nil, err
-	} else if cached != nil {
-		// "latest" caches which SHA is latest; Return the manifest for that SHA.
-		if sha == "latest" {
-			return getManifest(ctx, manifestAPI, string(cached.Value), paths)
+	if shared.IsLatest(sha) {
```

Comment:
Food for thought:  rewriting it [this way](https://play.golang.org/p/geYvFjHPvA4) can get rid of the nested `else`

Reply:
That looks better. Thanks!